packages/flutter/test/widgets/nested_scroll_view_test.dart

```diff
@@ -1906,6 +2165,115 @@ void main() {
       expect(tester.getCenter(find.text('Item 49')).dy, equals(585.0));
     }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS }));
   });
+
+  testWidgets('Scroll pointer signal should not cause overscroll.',
+      (WidgetTester tester) async {
+    final ScrollController controller = ScrollController();
+    await tester.pumpWidget(buildTest(controller: controller));
+
+    final Offset scrollEventLocation =
+        tester.getCenter(find.byType(NestedScrollView));
+    final TestPointer testPointer = TestPointer(1, ui.PointerDeviceKind.mouse);
+    // Create a hover event so that |testPointer| has a location when generating the scroll.
+    testPointer.hover(scrollEventLocation);
+
+    await tester
+        .sendEventToBinding(testPointer.scroll(const Offset(0.0, 20.0)));
+    expect(controller.offset, 20);
+
+    await tester
+        .sendEventToBinding(testPointer.scroll(const Offset(0.0, -40.0)));
+    expect(controller.offset, 0);
+
+    await tester.tap(find.text('DD'));
+    await tester.pumpAndSettle();
+
+    await tester
+        .sendEventToBinding(testPointer.scroll(const Offset(0.0, 1000000.0)));
+    expect(find.text('ddd1'), findsOneWidget);
+  });
+
+  testWidgets('Scroll pointer signal', (WidgetTester tester) async{
+    await tester.pumpWidget(buildTest());
+    expect(find.text('aaa2'), findsOneWidget);
+    expect(find.text('aaa3'), findsNothing);
+    expect(find.text('bbb1'), findsNothing);
+    await tester.pump(const Duration(milliseconds: 250));
+    expect(
+      tester.renderObject<RenderBox>(find.byType(AppBar)).size.height,
+      200.0,
+    );
+
+    final TestPointer testPointer = TestPointer(1, ui.PointerDeviceKind.mouse);
+    // The offset is the responsibility of innerPosition.
+    // Related to https://github.com/flutter/flutter/issues/55362
```

Comment:
We usually just add a single comment right at the top of the test
```suggestion
    // Regression test for https://github.com/flutter/flutter/issues/55362
```